run/test/specs/disappearing_message_checks_android.spec.ts

```diff
@@ -0,0 +1,477 @@
+    device2
+  );
+  // TODO FIX
+  // await device2.disappearingControlMessage(
+  //   `${userA.userName} has set messages to disappear ${time} after they have been ${mode}.`
+  // );
+  // await device2.disappearingControlMessage(
+  //   `You set messages to disappear ${time} after they have been ${mode}.`
+  // );
+  // Wait for control messages to disappear
+  // await sleepFor(60000);
+  await device1.longPress("New voice message");
+  await device1.clickOnByAccessibilityID("Continue");
+  await device1.clickOnElementXPath(
+    `/hierarchy/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.ScrollView/android.widget.LinearLayout/android.widget.LinearLayout/android.widget.LinearLayout[2]/android.widget.Button[1]`
+  );
+  await device1.pressAndHold("New voice message");
+  // await device1.clickOnByAccessibilityID("OK");
+  // await device1.pressAndHold("New voice message");
+  await device1.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Voice message",
+  });
+  await device2.clickOnByAccessibilityID("Untrusted attachment message");
+  await device2.clickOnByAccessibilityID("Download media");
+  await sleepFor(60000);
+  await device1.hasElementBeenDeletedNew({
+    strategy: "accessibility id",
+    selector: "Voice message",
+  });
+  await device2.hasElementBeenDeletedNew({
+    strategy: "accessibility id",
+    selector: "Voice message",
+  });
+  await closeApp(device1, device2);
+}
+
+async function disappearingGifMessage(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const time: DMTimeOption = "1 minute";
+  const mode: DisappearActions = "sent";
+  // Create user A and user B
+  const [userA, userB] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+  ]);
+  await newContact(platform, device1, userA, device2, userB);
+  await setDisappearingMessage(
+    platform,
+    device1,
+    ["1:1", "Disappear after send option", "1 minute"],
+    device2
+  ); // Wait for control messages to disappear before sending image (to check if the control messages are interfering with finding the untrusted attachment message)
+  // TODO FIX
+  // await device2.disappearingControlMessage(
+  //   `${userA.userName} has set messages to disappear ${time} after they have been ${mode}.`
+  // );
+  // await device2.disappearingControlMessage(
+  //   `You set messages to disappear ${time} after they have been ${mode}.`
+  // );
+  // await sleepFor(60000);
+  // Click on attachments button
+  await device1.clickOnByAccessibilityID("Attachments button");
+  // Select GIF tab
+  await device1.clickOnByAccessibilityID("GIF button");
+  await device1.clickOnElementAll({
+    strategy: "accessibility id",
+    selector: "Continue",
+  });
+  // Select gif
+  await sleepFor(500);
+  await device1.clickOnElementXPath(
```

Comment:
you can search for that function an make sure the xpath is part of a constant in testing.ts (where other xpaths are defined)